# Layoffs Analysis

In the past few months, the world have seen a strong wave hitting the world especially in the Tech industry. With companies like Twitter, Amazon, and Meta laying off more than 10,000 Employee each causing panic in the world's economy along with impacting more than 30,000 families across the globe.

I thought of understanding the layoffs happening all around the world including the following metrics:
- Which industries were impacted by the layoffs wave?
- Which countries layed off the most employees?
- Which companies layed off the most employees?
- The correlation between funding stage and layoffs?
- The correlation between the amount of fund raised to the company and the layoffs?
- The impact of dollar inflation on the layoffs happening throughout the past 3 years?

This report includes the full explanatory analysis against multiple datasets, we have 1 main dataset and 1 supplementary datasets to support the analysis. These datasets are:
- Layoffs data from 2019 up to 2022 (Main Dataset)
- Dollar inflation 1800-2022 (Supplementary Dataset)

In this report, we will be loading, exploring and cleaning the datasets we have then we will export the data to CSV files so we can import it on Tableau for data visualization purposes and extracting insights.

### Data Exploration and cleaning.

In this section, we will load our data to pandas dataframes and start exploring our options so we can properly clean the data.

In [62]:
#Loading necessary libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

#loading the layoffs data
df_layoffs = pd.read_csv('layoffs.csv')
df_layoffs.head()

company       location     industry  total_laid_off  \
0         Roku    SF Bay Area        Media           200.0   
1      Orchard  New York City  Real Estate           180.0   
2  Chili Piper  New York City        Sales            58.0   
3    Capitolis  New York City      Finance            37.0   
4     TealBook        Toronto        Other            34.0   

   percentage_laid_off        date     stage        country  funds_raised  
0                 0.07  2022-11-17       IPO  United States         208.0  
1                  NaN  2022-11-17  Series D  United States         472.0  
2                  NaN  2022-11-17  Series B  United States          54.0  
3                 0.25  2022-11-17  Series D  United States         281.0  
4                 0.19  2022-11-17  Series B         Canada          73.0

For the layoffs data we have the company name, location,Country, funding stage, funds raised in million dollarsand industry along with the total number of laid off employees, percentage laid off per date.

Let us explore the data types and the Null values.

In [63]:
df_layoffs.dtypes

company                 object
location                object
industry                object
total_laid_off         float64
percentage_laid_off    float64
date                    object
stage                   object
country                 object
funds_raised           float64
dtype: object

In [64]:
df_layoffs.isnull().sum()

company                  0
location                 0
industry                 3
total_laid_off         467
percentage_laid_off    541
date                     0
stage                    4
country                  0
funds_raised           112
dtype: int64

We have around 467 null values for the total_laid_off employees, 541 null values for the percentage_laid_off, 112 null values for the funds_raised, 4 in the funding stage and finally 3 null values for the industry.

I will drop the null values for the total_laid_off as an initial cleaning step.

In [65]:
df_layoffs = df_layoffs.dropna(axis=0, subset=['total_laid_off'])
df_layoffs.isnull().sum()

company                  0
location                 0
industry                 3
total_laid_off           0
percentage_laid_off    300
date                     0
stage                    1
country                  0
funds_raised            69
dtype: int64

Regarding the industry and the stage columns, I will replace the null values with Unknown.

In [66]:
df_layoffs[['stage','industry']] = df_layoffs[['stage','industry']].fillna('Unknown')
df_layoffs.isnull().sum()

company                  0
location                 0
industry                 0
total_laid_off           0
percentage_laid_off    300
date                     0
stage                    0
country                  0
funds_raised            69
dtype: int64

For the percentage laid off and the fund raised columns, I will replace the null values with the average of the column itself.

In [67]:
df_layoffs['percentage_laid_off'].fillna(value=df_layoffs['percentage_laid_off'].mean(),inplace=True)
df_layoffs['funds_raised'].fillna(value=df_layoffs['funds_raised'].mean(),inplace=True)
df_layoffs.isnull().sum()

company                0
location               0
industry               0
total_laid_off         0
percentage_laid_off    0
date                   0
stage                  0
country                0
funds_raised           0
dtype: int64

Now that we have cleaned the data from layoffs dataset, let us export the data to CSV then move to the next dataset.

In [68]:
df_layoffs.to_csv('Layoffs_cleaned.csv')

#### Exploring and Cleaning the Dollar Inflation Dataset.

In [30]:
#Loading the Dollar Inflation DataSet
df_inf = pd.read_csv('dollar_inflation.csv')
df_inf.head()

Year  Dollar Value  Buying Power Inflation Rate
0  1800          1.00          1.00           2.44
1  1801          1.01          0.99           0.79
2  1802          0.85          1.18         -15.75
3  1803          0.90          1.11           5.61
4  1804          0.94          1.06           4.42

For this dataset, we have the Dollar value, Buying power and Inflation Rate  percentage per year.

Before actually doing anything, I will drop all values for all years except for 2020 up to 2022 for proper analysis. 

In [31]:
df_inf.shape

(223, 4)

In [35]:
df_inf.tail()

Year  Dollar Value  Buying Power Inflation Rate
218  2018         19.94          0.05           2.49
219  2019         20.29          0.05           1.76
220  2020         20.54          0.05           1.23
221  2021         21.51          0.05           4.70
222  2022         23.56          0.04        #VALUE!

In [ ]:
df_inf = df_inf[220: ]

In [42]:
df_inf

Year  Dollar Value  Buying Power Inflation Rate
220  2020         20.54          0.05           1.23
221  2021         21.51          0.05           4.70
222  2022         23.56          0.04        #VALUE!

We have a missing value for the Inflation rate in 2022. This data is collected up to November 2022.
According to the International Monetary Fund the Dollar Inflation rate as per Ocotber 2022 is 7.75%. Therefore, I will replace the missing value with the number announced by IMF.

In [43]:
df_inf['Inflation Rate'].fillna(value=7.75,inplace=True)
df_inf

C:\Users\ahmed\AppData\Local\Temp\ipykernel_7396\388052671.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_inf['Inflation Rate'].fillna(value=7.75,inplace=True)


Year  Dollar Value  Buying Power Inflation Rate
220  2020         20.54          0.05           1.23
221  2021         21.51          0.05           4.70
222  2022         23.56          0.04        #VALUE!

In [44]:
df_inf.dtypes

Year                int64
Dollar Value      float64
Buying Power      float64
Inflation Rate     object
dtype: object

In [55]:
df_inf['Inflation Rate'] = df_inf['Inflation Rate'].replace('#VALUE!',7.75)
df_inf.to_csv('Inflation Rate Updated.csv')

C:\Users\ahmed\AppData\Local\Temp\ipykernel_7396\4067878897.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_inf['Inflation Rate'] = df_inf['Inflation Rate'].replace('#VALUE!',7.75)


In [77]:
df_layoffs.to_csv('Layoffs_cleaned.csv')

### Data Wrangling and Visualization

I have exported the cleaned data to csv files so I can visualize it using Tableau and re-embed the Visuals Here.

As per the analysis, it looks like **United States and India** recorded the highest number of layoffs in the last 3 years (2020 to 2021) as follows:

In [108]:
%%html
<div class='tableauPlaceholder' id='viz1671132305832' style='position: relative'><noscript><a href='#'><img alt='Top Countries Laid off Employees ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerCountry&#47;Sheet1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TechLayoffsPerCountry&#47;Sheet1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerCountry&#47;Sheet1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671132305832');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

The number of layoffs in the **United States** is pretty huge, so let us take a deeper snapshot on the layoffs in the US specifically.

In the United States, San Francisco Bay Area recorded the highest number of layoffs in 2020 to 2022 with total number of 75,685 laid off employee in the tech industry followed by New York with total of 21,221 laid of employee as follows:

In [109]:
%%html
<div class='tableauPlaceholder' id='viz1671132329989' style='position: relative'><noscript><a href='#'><img alt='Layoffs Per Location in US ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsLocationinUS&#47;Sheet2&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TechLayoffsLocationinUS&#47;Sheet2' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsLocationinUS&#47;Sheet2&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671132329989');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

I needed to understand which industries had the most layoffs in the US Tech market and it was detected that the following are the top industries in terms of Layoffs:
- Consumer (23,002 laid off employee in the last 3 years)
- Retail (20,203)
- Transportation (16,510)
- Real Estate (13,912)

In [110]:
%%html
<div class='tableauPlaceholder' id='viz1671132614357' style='position: relative'><noscript><a href='#'><img alt='Layoffs Per Industry in US ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerIndustryinUS&#47;Sheet3&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TechLayoffsPerIndustryinUS&#47;Sheet3' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerIndustryinUS&#47;Sheet3&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671132614357');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Going even further, we need to know which companies performed the most layoffs in the United States and it appears that **Meta, Amazon & Uber** have recorded the highest number of layoffs over the past 3 years knowing that Meta and Amazon started laying off employees in 2022 only!! 

In [111]:
%%html
<div class='tableauPlaceholder' id='viz1671132896368' style='position: relative'><noscript><a href='#'><img alt='Layoffs Per Company in US ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerCompanyinUS&#47;Sheet4&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='TechLayoffsPerCompanyinUS&#47;Sheet4' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Te&#47;TechLayoffsPerCompanyinUS&#47;Sheet4&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /><param name='ignore_sticky_session' value='yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671132896368');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

### Understanding the reason behind the layoffs

In this section, I will try to understand the reason behind the layoffs hitting the tech market but first I will exclude United States and India as they are considered outliers in the dataset then I will do the following:
- Define the correlation between the Funds Raised and the layoffs.
- Define the impact of the Dollar inflation & Buying Power on the Tech market.

#### Correlation between Funds Raised and Layoffs

In [112]:
%%html
<div class='tableauPlaceholder' id='viz1671133095997' style='position: relative'><noscript><a href='#'><img alt='Correlation between Fund Raised &amp; Layoffs Excluding United States and India. ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Co&#47;CorrelationbetweenFundsRaisedTechLayoffs&#47;Sheet5&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='CorrelationbetweenFundsRaisedTechLayoffs&#47;Sheet5' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Co&#47;CorrelationbetweenFundsRaisedTechLayoffs&#47;Sheet5&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /><param name='ignore_sticky_session' value='yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671133095997');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

It is pretty clear that Funds raised and layoffs has a positive correlation which means that when companies raise more funds they tend to layoff more people and I believe this is obligations from the shareholders/investors.

#### Impact of Dollar Inflation

The dollar inflation has really increased in the past 3 years as an impact left by COVID-19 followed by the War between Russia and Ukraine. But let us try to understand whether the inflation impacts the layoffs or not?

In [113]:
%%html
<div class='tableauPlaceholder' id='viz1671133303798' style='position: relative'><noscript><a href='#'><img alt='Relation Between Dollar Inflation and Layoffs ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Re&#47;RelationbetweenDollarInflationandLayoffs&#47;Sheet7&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='RelationbetweenDollarInflationandLayoffs&#47;Sheet7' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Re&#47;RelationbetweenDollarInflationandLayoffs&#47;Sheet7&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en-US' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1671133303798');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

In the above figure, it looks like the dollar inflation has no impact on the layoffs since it kept increasing year over year but the layoffs actually dropped in 2021 but that is not right. 

Dollar inflation is not an enough measure, but if we look at the buying power we will get a better understanding:
- From 2020 - 2021, the buying power remained the same although the dollar inflation increased which explains why the layoffs decreased from 2020 to 2021

- From 2021 - 2022, the buying power decreased by around 10% and the dollar inflation increased as well which explains why the layoffs made a spike by this time.

## Conclusion

It looks like there are two main factors affecting the tech layoffs:
- Dollar inflation and Buying Power in relevance to each other
- Funds being raised without proper planning

There might be another reason why the tech market was hit by this wave of layoffs which is the shareholders directing their investment to another industries like Oil & Gas.

In a nut shell, companies need to stop over utilizing their teams and plan for proper funding so that they can keep their employees safe and happy rather than laid off. Also, we need to understand why the shareholders are shifting their investment to other industries.